packages/api/buildworkermodule.mjs

```diff
@@ -1,23 +1,69 @@
-import path from 'path'
-import { fileURLToPath } from 'url'
-import { build } from 'esbuild'
+import fs from "fs";
+import path from "path";
+import { fileURLToPath } from "url";
+import { build } from "esbuild";
+import git from "git-rev-sync";
+import Sentry from "@sentry/cli";
 
-const __filename = fileURLToPath(import.meta.url)
-const __dirname = path.dirname(__filename)
+(async () => {
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+  const pkg = JSON.parse(
+    fs.readFileSync(path.join(__dirname, "package.json"), "utf8")
+  );
 
-try {
-  build({
-    bundle: true,
-    sourcemap: true,
-    format: 'esm',
-    target: 'esnext',
-    external: ['__STATIC_CONTENT_MANIFEST'],
-    conditions: ['worker', 'browser'],
-    entryPoints: [path.join(__dirname, 'src', 'index.ts')],
-    outdir: path.join(__dirname, 'dist'),
-    outExtension: { '.js': '.mjs' }
-  })
-} catch(err) {
-  process.exitCode = 1
-  throw new Error(err)
-}
+  try {
+    // Release name cannot contain slashes, and are global per org, so we use
+    // custom prefix here not pkg.name.
+    // See https://docs.sentry.io/platforms/javascript/guides/cordova/configuration/releases/
+    const sentryRelease = `w3name-api@${pkg.version}+${git.short(__dirname)}`;
+    console.log(`Building ${sentryRelease}`);
+
+    await build({
+      bundle: true,
+      sourcemap: true,
+      format: "esm",
+      target: "esnext",
+      external: ["__STATIC_CONTENT_MANIFEST"],
+      conditions: ["worker", "browser"],
+      entryPoints: [path.join(__dirname, "src", "index.ts")],
+      outdir: path.join(__dirname, "dist"),
+      outExtension: { ".js": ".mjs" },
+      define: {
+        SENTRY_RELEASE: JSON.stringify(sentryRelease),
+        VERSION: JSON.stringify(pkg.version),
+        COMMITHASH: JSON.stringify(git.long(__dirname)),
+        BRANCH: JSON.stringify(git.branch(__dirname)),
+        global: "globalThis",
+      },
+    });
+
+    // Sentry release and sourcemap upload
+    if (process.env.SENTRY_UPLOAD === "true") {
+      console.log(`Uploading to Sentry ${sentryRelease}`);
+      const cli = new Sentry(undefined, {
+        authToken: process.env.SENTRY_TOKEN,
+        org: "protocol-labs-it",
+        project: "w3name-api",
+      });
+
+      await cli.releases.new(sentryRelease);
+      await cli.releases.setCommits(sentryRelease, {
+        auto: true,
+        ignoreEmpty: true,
+        ignoreMissing: true,
+      });
+      await cli.releases.uploadSourceMaps(sentryRelease, {
+        // validate: true,
+        include: [path.join(__dirname, "..", "dist")],
```

Comment:
From doing a comparison to the same code in web3.storage, by my calculation the `'..'` should be omitted here. This script lives in `packages/api/`, whereas the script which contains the exact same line of code in web3.storage lives in `packages/api/scripts/`. And `__dirname` is defined in the same way in both.

Reply:
Yup, you're right. Corrected this. Good spot.